frontend/src/components/common/Grid/Grid.tsx

```diff
@@ -0,0 +1,83 @@
+import styled from 'styled-components';
+
+import GridItem from './GridItem';
+
+const formatAreas = (areas: string[]) => areas.map(area => `'${area}'`).join(' ');
+
+const frGetter = (value: string | number) =>
+  typeof value === 'number' ? `repeat(${value}, 1fr)` : value;
+
+const gap = ({ gap = '0.8rem' }: IGridProps) => gap;
+
+export interface IGridProps {
+  inline?: boolean;
+  height?: string;
+  columns?: string | number;
+  rows?: string;
+  gap?: string;
+  columnGap?: string;
+  rowGap?: string;
+  areas?: string[];
+  justifyContent?: string;
+  alignContent?: string;
+}
+
+/**
+ * A thin wrapper to help make CSS Grid simpler and more expressive in React
+ *
+ * Basic Usage
+ *
+ *  <Grid columns='10px 20px 30px'>
+ *    <MyComponent />  # assumes MyComponent is aware of how to position itself within the parent grid
+ *  </Grid>
+ *
+ * ADVANCED Usage - to help with structuring your components, a Grid Item is also available.
+ *
+ *  <Grid columns="100px 1fr 100px" rows="50px auto 25px" areas={[
+ *    'header header header',
+ *    'sidebar content ads',
+ *    'footer footer footer',
+ *  ]}>
+ *    <Grid.Item area="header">
+ *      <MyHeader />
+ *    </Grid.Item>
+ *
+ *    <Grid.Item area="sidebar">
+ *      <MySidebar />
+ *    </Grid.Item>
+ *
+ *    <Grid.Item area="content">
+ *      <MyContentArea />
+ *    </Grid.Item>
+ *
+ *    <Grid.Item area="ads">
+ *      <MyAdsBanner />
+ *    </Grid.Item>
```

Comment:
q: Is this component based on a react component? What are the benefits of it over the bootstrap <Row> <Col>?

Reply:
1. React-Bootstrap doesn't provide a Grid component (that I know of)
2. We have been moving to CSS Grid from Flexbox in newer components - however we are copy/pasting a lot when we could create a generic Grid component
3. You are right that is not being used because I though I would in my code but ended up not doing it to avoid refactoring existing grids (such as FormGrid)
4. Sets the table for such refactoring (if needed). If not, you can use it if you find it simpler to mucking with CSS/styled components directly. Up to the dev

Reply:
Also note this would be for simple cases. I did not include all available CSS Grid properties here.
You can always add you custom grid prop like so:

```tsx
const MyGrid = styled(Grid)`
  some-grid-prop: some-value;
`
```

Reply:
sorry, I think the last part of my question didn't render `<Row> <Col>`.
It is not being used, should we hold on to it? As it is right now is dead code. We could re-introduce it if necessary. Also, a demo would be cool!

Reply:
I see, `<Row>` and `<Col>` are based on Flexbox. We have already tons of forms on Flexbox but decided in a dev meeting to start migrating towards CSS grid because:

1. It's less "verbose" and more expressive (ie you don't need a ton of Rows, Cols to achieve complex layouts)
2. It's supported by major browsers now so it's kinda safe to use CSS grid

Reply:
Why not use the package instead of just copying the source? https://www.npmjs.com/package/@react-css/grid?activeTab=readme

Reply:
Will remove and replace with npm package when needed